From ad208d03a5a32957a419eb1a473aa96ff4da45ca Mon Sep 17 00:00:00 2001
From: Pavel Machek <pavel@denx.de>
Date: Sat, 27 Oct 2012 01:00:53 +0200
Subject: [PATCH 242/951] ARM: socfpga: fix build break for allyesconfig

Building a multiplatform kernel with combined armv7 and armv6
kernel does not work with the socfpga smp code, which contains
v7-only instructions. Since this code never runs on armv6,
we can annotate the assmebly with .arch   armv7-a, as suggested
by Russell King.

[arnd: reword commit message]

Signed-off-by: Pavel Machek <pavel@denx.de>
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
---
 arch/arm/mach-socfpga/headsmp.S | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-socfpga/headsmp.S b/arch/arm/mach-socfpga/headsmp.S
index 17d6eaf..65762ab 100644
--- a/arch/arm/mach-socfpga/headsmp.S
+++ b/arch/arm/mach-socfpga/headsmp.S
@@ -11,6 +11,7 @@
 #include <linux/init.h>
 
 	__INIT
+	.arch	armv7-a
 
 #define CPU1_START_ADDR 	        0xffd08010
 
-- 
1.8.0

